fastify.js

```diff
@@ -344,7 +344,7 @@ function build (options) {
     throwIfAlreadyStarted('Cannot call "addHook" when fastify instance is already started!')
 
     // TODO: v3 instead of log a warning, throw an error
-    if (name === 'onSend' || name === 'preSerialization') {
+    if (name === 'onSend' || name === 'preSerialization' || name === 'onError') {
```

Comment:
What's reason to check the name at all? Isn't enough to check for argument length of 3 and 4?

Reply:
The `onError` hook can have either 4 or 3 arguments depending if it's async or not
https://github.com/fastify/fastify/blob/master/docs/Hooks.md#onerror

In this PR I just added `onError` to the same place where this is being checked for `onSend` and `preSerialization`. If it's async then it should have max 4 arguments, 3 otherwise.

Reply:
No, what I'm trying to say is:
```js
    if (fn.constructor.name === 'AsyncFunction') {
      // all hooks have either 3 or 4 arguments and it's the callback
      if (fn.length >= 3) {
        fastify.log.warn("Async function has too many arguments. Async hooks should not use the 'next' argument.", new Error().stack)
      }
    }
```
would archive the same right?

Reply:
No, that would still trigger the warning for `onError`, `onSend` and `preSerialization` as these hooks have 3 arguments in their async version so we need to exclude them from this check or to increase the maximum argument count check to 4 (as in the PR)

Reply:
I got it.